### PR TITLE
2018-2019 school demographics stats import

### PR DESCRIPTION
**Pull request by bencodeorg**
Script to import demographic stats about public and charter schools for the 2018-2019 school year. Very similar to this [script](https://github.com/code-dot-org/code-dot-org/blob/staging/bin/oneoff/nces_data/import_school_stats_by_years_2017_2018_ccd) and [PR](https://github.com/code-dot-org/code-dot-org/pull/31741), which was used to import this information about a year ago for the 2017-2018 school year.

## Testing story

Executed each file import after I wrote it (adding each to the bottom of the file subsequently) via `./import_school_stats_by_years_2018_2019_ccd`. I confirmed that the values that were inserted into the `school_stats_by_years` table were sane by connecting to the table in Tableau and looking at histograms of the values in each column.

## Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+def sanitize_string_for_db(unsanitized)
+  unsanitized = unsanitized == '-' ? nil : unsanitized
+  unsanitized&.tr('="', '')
+end
+
+AWS::S3.process_file('cdo-nces', "#{SURVEY_YEAR}/ccd/ccd_sch_029_1819_w_1a_091019.csv") do |filename|
+  SchoolStatsByYear.transaction do
+    SchoolStatsByYear.merge_from_csv(filename, {col_sep: ",", headers: true, quote_char: "\x00", encoding: 'ISO-8859-1:UTF-8'}) do |row|
+      {
+        school_id:          row['NCESSCH'].to_i.to_s,
```
Comment (jamescodeorg) [inline bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd:58]: Why do we convert to integer and then convert back to string?

File: bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+AWS::S3.process_file('cdo-nces', "#{SURVEY_YEAR}/ccd/School_lvl_demo.csv") do |filename|
+  SchoolStatsByYear.transaction do
+    SchoolStatsByYear.merge_from_csv(filename, {col_sep: ",", headers: true, quote_char: "\x00"}) do |row|
+      row = row.to_h.map {|k, v| [k, sanitize_string_for_db(v)]}.to_h
+
+      {
+        school_id:      row['School ID - NCES Assigned [Public School] Latest available year'].to_i.to_s,
+        school_year:    SURVEY_YEAR,
+
+        students_total:     row['Total Students All Grades (Includes AE) [Public School] 2018-19'].presence.try {|v| v.to_i == 0 ? nil : v.to_i},
```
Comment (jamescodeorg) [inline bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd:124]: Why do we need the extra logic here?

**Review by jamescodeorg — COMMENTED** [review]
Script looks good to me.  I think it might be helpful to include some information about where to find the csv files on the NCES website and/or a link to Liz's doc.

Also, I didn't see error handling so I assume that all of the data is processed cleanly?

File: bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd (hunk 1 above)
Comment (bencodeorg) [reply bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd:58]: Ahh great question. At one point, our schools table was only public and charter schools, which have IDs that are all numbers, but have leading zeros. They were imported and coerced to integers, so "012" became 12. Then, we added private schools, which have letters in the IDs (at which point, we moved the ID field in the schools table back to a string). So, this hack turns that "012" into 12, then back to "12" 🤦 .

File: bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd (hunk 2 above)
Comment (bencodeorg) [reply bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd:124]: Schools presumably shouldn't have 0 total students -- not sure this is definitely the right approach, but is at least consistent with the prior import.

**Comment by bencodeorg** [comment]
> Script looks good to me. I think it might be helpful to include some information about where to find the csv files on the NCES website and/or a link to Liz's doc.
> 
> Also, I didn't see error handling so I assume that all of the data is processed cleanly?

Will do re: adding more info about where to find the raw files.

Very minimal error handing is [here](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/school_stats_by_year.rb#L60) -- in my testing, I'd see the occasional school get skipped (maybe less than 100 in the 100K rows imported, which is kind of expected based on previous years).

I also don't stress too much about this import because it's primarily used by our data team for analytics work (not a ton of use by our application), and we're just adding a bunch of new rows (there are no rows in the table currently for the 2018-2019 school year) instead of updating existing ones, so could easily delete bad data if something was wrong.

File: bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd (hunk 2 above)
Comment (jamescodeorg) [reply bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd:124]: Ok, I'm not sure if I'm reading this right or not, but it seems like this line just turns a 0 to a nil.  This is ok, but then I'm curious why we didn't do this for all of the other values.  At any rate, leaving as-is to be consistent with prior imports sounds good.

**Review by jamescodeorg — APPROVED** [review]
Good to merge after adding a comment about where to get the data.

File: bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd (hunk 2 above)
Comment (bencodeorg) [reply bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd:124]: Yeah I think you're reading it right -- I think the thought it that it's plausible for a school to have, say, 0 Hawaiian/Pacific Islander students, but not plausible to have 0 total students. So we wipe total students but not sub-groups.

**Pull request merged**